edx_sga/sga.py

```diff
@@ -0,0 +1,530 @@
+    """
+    This block defines a Staff Graded Assignment.  Students are shown a rubric
+    and invited to upload a file which is then graded by staff.
+    """
+    has_score = True
+    icon_class = 'problem'
+
+    display_name = String(
+        default='Staff Graded Assignment', scope=Scope.settings,
+        help="This name appears in the horizontal navigation at the top of "
+             "the page."
+    )
+
+    weight = Float(
+        display_name="Problem Weight",
+        help=("Defines the number of points each problem is worth. "
+              "If the value is not set, the problem is worth the sum of the "
+              "option point values."),
+        values={"min": 0, "step": .1},
+        scope=Scope.settings
+    )
+
+    points = Integer(
+        display_name="Maximum score",
+        help=("Maximum grade score given to assignment by staff."),
+        default=100,
+        scope=Scope.settings
+    )
+
+    staff_score = Integer(
+        display_name="Score assigned by non-instructor staff",
+        help=("Score will need to be approved by instructor before being "
+              "published."),
+        default=None,
+        scope=Scope.settings
+    )
+
+    comment = String(
+        display_name="Instructor comment",
+        default='',
+        scope=Scope.user_state,
+        help="Feedback given to student by instructor."
+    )
+
+    annotated_sha1 = String(
+        display_name="Annotated SHA1",
+        scope=Scope.user_state,
+        default=None,
+        help=("sha1 of the annotated file uploaded by the instructor for "
+              "this assignment.")
+    )
+
+    annotated_filename = String(
+        display_name="Annotated file name",
+        scope=Scope.user_state,
+        default=None,
+        help="The name of the annotated file uploaded for this assignment."
+    )
+
+    annotated_mimetype = String(
+        display_name="Mime type of annotated file",
+        scope=Scope.user_state,
+        default=None,
+        help="The mimetype of the annotated file uploaded for this assignment."
+    )
+
+    annotated_timestamp = DateTime(
+        display_name="Timestamp",
+        scope=Scope.user_state,
+        default=None,
+        help="When the annotated file was uploaded")
+
+    def max_score(self):
+        return self.points
+
+    @reify
+    def block_id(self):
+        # cargo culted gibberish
+        return self.scope_ids.usage_id
+
+    def student_submission_id(self, id=None):
+        """
+        Returns dict required by the submissions app for creating and
+        retrieving submissions for a particular student.
+        """
+        if id is None:
+            id = self.xmodule_runtime.anonymous_student_id
+            assert id != 'MOCK', "Forgot to call 'personalize' in test."
+        return {
+            "student_id": id,
+            "course_id": self.course_id,
+            "item_id": self.block_id,
+            "item_type": 'sga',  # ???
```

Comment:
"sga" makes sense to me. For ORA2 it is "openassessment" and it's just primarily for identifying the origin of the student item. 

This dictionary required by the submission API is the serialized representation of the StudentItem model, which is also part of the edx-submissions project.